PROTOCOL.md

```diff
@@ -710,25 +807,39 @@ Writers should reject any transaction that contains data where the expression `x
 
 ## CHECK Constraints
 
+Enablement:
+- If the table is on a Writer Version up to 6, CHECK Constraints is enabled when the table's metadata contains required keys.
+- If the table is on Writer Version 7, a feature name `checkConstraints` must exist in the table `protocol`'s `writerFeatures`.
+
 CHECK constraints are stored in the map of the `configuration` field in [Metadata](#change-metadata). Each CHECK constraint has a name and is stored as a key value pair. The key format is `delta.constraints.{name}`, and the value is a SQL expression string whose return type must be `Boolean`. Columns referred by the SQL expression must exist in the table schema.
 
 Rows in a table must satisfy CHECK constraints. In other words, evaluating the SQL expressions of CHECK constraints must return `true` for each row in a table.
 
 For example, a key value pair (`delta.constraints.birthDateCheck`, `birthDate > '1900-01-01'`) means there is a CHECK constraint called `birthDateCheck` in the table and the value of the `birthDate` column in each row must be greater than `1900-01-01`.
 
 Hence, a writer must follow the rules below:
+- Upon the first time a writer add CHECK Constraints to a table and if the table is on Writer Version 7, the writer must write a `protocol` action to add a feature name `checkConstraints` to `writerFeatures`.
```

Comment:
I would just say that the writer cannot do this unless the feature is enabled (according to the enablement rules above). And that the writer may enable the feature in the same commit where they make use of the feature for the first time.

Make sure this message is consistent across all of the features.